module-apps/application-call/model/CallModel.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "CallModel.hpp"
+
+#include <service-cellular/service-cellular/CellularServiceAPI.hpp>
+#include <service-evtmgr/service-evtmgr/EVMessages.hpp>
+#include <service-evtmgr/Constants.hpp>
+
+namespace app::call
+{
+    time_t CallModel::getTime()
+    {
+        return callTime;
+    }
+
+    void CallModel::setTime(time_t newTime)
+    {
+        callTime = newTime;
+
+        if (notifyPresenterOnDurationChange != nullptr) {
+            notifyPresenterOnDurationChange();
+        }
+    }
+
+    CallState CallModel::getState()
+    {
+        return callState;
+    }
+
+    void CallModel::setState(CallState newState)
+    {
+        if (callState == newState) {
+            LOG_INFO("Dropping call state change");
+        }
+
+        if (callState == CallState::Incoming && newState == CallState::Ended && callWasRejected) {
+            callWasRejected = false;
+            callState       = CallState::Rejected;
+        }
+        else {
+            callState = newState;
+        }
+
+        if (callState == CallState::Active) {
+            turnOnKeypadBacklight();
+        }
+        else {
+            turnOffKeypadBacklight();
+        }
```

Comment:
while setting the model state is ok - turning stuff on and off less so ( I know that this is mostly moved logic here and do not request to change it all - I'm just mentioning it as I go on with the review :) )